### PR TITLE
Introduce `IsIdentityOperation` matcher for use by Refaster templates

### PR DESCRIPTION
**Pull request by Stephan202**
Suggested commit message:
```
Introduce `IsIdentityOperation` matcher for use by Refaster templates (#749)
```

### CODE REVIEW

File: error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ComparatorRulesTestInput.java [hunk 1]
```diff
@@ -31,6 +31,7 @@ ImmutableSet<Comparator<String>> testNaturalOrder() {
         String::compareTo,
         Comparator.comparing(identity()),
         Comparator.comparing(s -> s),
+        Comparator.comparing(s -> 0),
```
Comment (Stephan202) [inline error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ComparatorRulesTestInput.java:34]: For each rule, added a non-identity transformation test case, to validate that the `@Matcher` annotation is present.

File: refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/IsIdentityOperationTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            "",
+            "  IntUnaryOperator positive4() {",
+            "    // BUG: Diagnostic contains:",
+            "    return IntUnaryOperator.identity();",
+            "  }",
+            "",
+            "  LongUnaryOperator positive5() {",
+            "    // BUG: Diagnostic contains:",
+            "    return LongUnaryOperator.identity();",
+            "  }",
```
Comment (Stephan202) [inline refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/IsIdentityOperationTest.java:51]: Previously we would only match `Function.identity()`. Our internal code base contains a number of `UnaryOperator.identity()` usages that were thus not matched by the Refaster rules.

**Review by Stephan202 — COMMENTED** [review]
Some context.

**Comment by Stephan202** [comment]
Rebased and resolved conflicts.

**Comment by Stephan202** [comment]
Rebased and resolved conflicts.

**Comment by Stephan202** [comment]
Generalized a few more Refaster rules this way.

**Review by Venorcis — APPROVED** [review]
LGTM 🚀

**Comment by Stephan202** [comment]
Rebased and resolved conflicts.

**Review by rickie — APPROVED** [review]
Nice one ✨ ! Love the improvements in the rule collections 😄 !

**Comment by rickie** [comment]
> Generalized a few more Refaster rules this way.

Want to add something to the commit message? Something like `While there, generalized Refaster rules using this matcher.`?

**Comment by Stephan202** [comment]
Good point @rickie. How about (slightly tweaked):
```
Introduce `IsIdentityOperation` matcher for use by Refaster templates (#749)

While there, generalized a number of Refaster rules using this new matcher.
```

**Comment by Stephan202** [comment]
I'll have a look at the conflict.

**Comment by Stephan202** [comment]
Conflict was trivial. Rebased.

**Pull request merged**